optuna/trial.py

```diff
@@ -394,19 +394,28 @@ def report(self, value, step=None):
 
     def should_prune(self, step=None):
         # type: (Optional[int]) -> bool
-        """Judge whether the trial should be pruned.
+        """Suggest whether the trial should be pruned or not.
 
-        This method calls prune method of the pruner, which judges whether the trial should
-        be pruned at the given step. Please refer to the example code of
-        :func:`optuna.trial.Trial.report`.
+        The suggestion is made by a pruning algorithm associated with the trial and is based on
+        previously reported values. The algorithm can be specified when constructing a
+        :class:`~optuna.study.Study`.
+
+        .. note::
+            If no values have been reported, the algorithm cannot make meaningful suggestions.
+            Similarly, if this method is called multiple times with the exact same set of reported
+            values, the suggestions will be the same.
```

Comment:
"will be" might be too strong as it's up to each pruner, unless we make it a requirement.

Reply:
I agree with you. In addition to your concern, if the optimization is being executed in a distributed manner, there is no guarantee that the suggestions will be the same because the state of the study can be updated anytime by other workers.